### PR TITLE
Create an array for query params with the same key

### PR DESCRIPTION
**Pull request by deirdreg**
Query string parameters with the same key are now stored as an array in the queryParameters dictionary instead of being overwritten with the next value.

### CODE REVIEW

File: DeepLinkKit/Categories/NSString+DPLQuery.m [hunk 1]
```diff
@@ -15,17 +15,26 @@ + (NSString *)DPL_queryStringWithParameters:(NSDictionary *)parameters {
...
 - (NSDictionary *)DPL_parametersFromQueryString {
-    NSArray *params = [self componentsSeparatedByString:@"&"];
-    NSMutableDictionary *paramsDict = [NSMutableDictionary dictionaryWithCapacity:[params count]];
-    for (NSString *param in params) {
-        NSArray *pairs = [param componentsSeparatedByString:@"="];
-        if ([pairs count] == 2) {
-            NSString *key   = [pairs[0] DPL_stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-            NSString *value = [pairs[1] DPL_stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-            paramsDict[key] = value;
+  NSArray *params = [self componentsSeparatedByString:@"&"];
```
Comment (chrismaddern) [inline DeepLinkKit/Categories/NSString+DPLQuery.m]: 4 character tabs :wink:

**Comment by deirdreg** [comment]
Updated to use 4 space tabs

**Comment by chrismaddern** [comment]
Hey @deirdreg - this is definitely something we want to support, a couple outstanding issues:

1. It'd be great to see a test to know that it doesn't regress.

2. Currently it won't support create `DPLDeepLink` objects with array parameters (https://github.com/myfitnesspal/DeepLinkKit/blob/develop/DeepLinkKit/Categories/NSString%2BDPLQuery.m#L9) - i.e. `initWithURL:` and `url` are no longer symmetrical.

**Comment by wessmith** [comment]
@deirdreg Do you have interest in completing this? If so, the PR should be updated with the latest from master, support for serializing array values back into query params, and tests to support the changes. Thanks!

**Pull request closed without merging**